### PR TITLE
Breakup hyperv machine funcs

### PR DESCRIPTION
**Pull request by jakecorrenti**
The functions for HyperV's VM interface implementation (machine.go) had quite large functions. Pulls out some code that could be moved to its own function for easier readability.

[NO NEW TESTS NEEDED]

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/machine/hyperv/machine.go [hunk 1]
```diff
@@ -67,6 +67,31 @@ type HyperVMachine struct {
 	LastUp time.Time
 }
 
+func addSSHConnectionsToPodmanSocket(m *HyperVMachine, opts machine.InitOptions) error {
```
Comment (vrothberg) [inline pkg/machine/hyperv/machine.go:85]: We can turn it into a method (with comment) instead:
```golang
func (m *HyperVMachine) add... error {}
```

File: pkg/machine/hyperv/machine.go [hunk 2]
```diff
@@ -67,6 +67,21 @@ type HyperVMachine struct {
 	LastUp time.Time
 }
 
+func addNetworkAndReadySocketsToRegistry(m *HyperVMachine) error {
```
Comment (vrothberg) [inline pkg/machine/hyperv/machine.go:70]: Same here: method with comments.

**Review by vrothberg — COMMENTED** [review]
Please make use of the type system as much as possible and use methods instead of functions with an IN/OUT parameter.

`git rebase -i ...` will avoid squashing all commits.

File: pkg/machine/hyperv/machine.go (hunk 1 above)
Comment (jakecorrenti) [reply pkg/machine/hyperv/machine.go]: Should be all set now

**Review by vrothberg — APPROVED** [review]
LGTM

**Comment by vrothberg** [comment]
@rhatdan @ashley-cui @baude PTAL

**Review by ashley-cui — COMMENTED** [review]
LGTM

**Review by vrothberg — APPROVED** [review]
/lgtm

**Pull request merged**